### PR TITLE
Split ykshim_client into production and testing modules.

### PR DESCRIPTION
**Pull request by vext01**
As suggested by @bjorn3 it probably makes sense to split ykshim_client into two parts (production and testing), as we did with the internal workspace.

The second commit does some renaming to make it clear what shim symbols are for what purpose.

### CODE REVIEW

File: ykshim_client/src/test_api.rs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        referenced: bool,
+    );
+    fn __ykshimtest_tracecompiler_local_to_location_str(
+        tc: *mut RawTraceCompiler,
+        local: Local,
+    ) -> *mut c_char;
+    fn __ykshimtest_tracecompiler_local_dead(tc: *mut RawTraceCompiler, local: Local);
+    fn __ykshimtest_tracecompiler_find_sym(sym: *mut c_char) -> *mut c_void;
+    fn __yktest_interpret_body(body_name: *mut c_char, icx: *mut u8);
+    fn __yktest_reg_pool_size() -> usize;
```
Comment (bjorn3) [inline ykshim_client/src/test_api.rs:43]: Why did you mix `__ykshimtest` and `__yktest`?

Comment (vext01) [reply ykshim_client/src/test_api.rs:43]: That's a mistake on my part. It's supposed to be `__ykshimtest` across the board. Will fix.

Comment (vext01) [reply ykshim_client/src/test_api.rs]: Fixed in 124cd35.

**Review by bjorn3 — APPROVED** [review]

**Comment by ptersilie** [comment]
Does this need squashing?

**Comment by vext01** [comment]
squashed.

**Pull request merged**